File: src/Backend/Common/Models/Seasons/DiscountPolicy.cs
using System;
using System.Collections.Generic;

namespace Company.Common.Models.Seasons
{
    public class DiscountPolicy
    {
        public int Id { get; set; }

        public int SeasonListValueId { get; set; }

        public DateTime CreateTime { get; set; }

        public int CreatorId { get; set; }

        public ICollection<SalesPlanData> SalesPlanDatas { get; set; } = new List<SalesPlanData>() { };

        public ICollection<BrandPolicyData> BrandPolicyDatas { get; set; } = new List<BrandPolicyData>() { };

        public ICollection<PolicyData> PolicyDatas { get; set; } = new List<PolicyData>() { };

        public virtual ICollection<ExchangeRates> ExchangeRates { get; set; }
    }
}

File: src/Frontend/projects/core/src/lib/models/dto/seasons/discount-policy.ts
import { BrandPolicyData } from './brand-policy-data';
import { ExchangeRates } from './exchange-rates';
import { PolicyData } from './policy-data';
import { SalesPlanData } from './sales-plan-data';

export class DiscountPolicy {
    public id: number;
    public seasonListValueId: number;
    public createTime: Date;
    public brandPolicyDatas: BrandPolicyData[] = [];
    public policyData: PolicyData;
    public salesPlanData: SalesPlanData;
    public exchangeRates: ExchangeRates;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }
        return Object.assign(
            new DiscountPolicy(),
            obj,
            {
                createTime: obj.createTime ? new Date(obj.createTime) : null,
                brandPolicyDatas: obj.brandPolicyDatas ? obj.brandPolicyDatas.map((bp) => BrandPolicyData.fromJSON(bp)) : null,
                policyData: obj.policyData ? PolicyData.fromJSON(obj.policyData) : null,
                salesPlanData: obj.salesPlanData ? SalesPlanData.fromJSON(obj.salesPlanData) : null,
                exchangeRates: obj.exchangeRates ? ExchangeRates.fromJSON(obj.exchangeRates) : null,
            },
        );
    }
}
